tools/matc/src/matc/ParametersProcessor.cpp

```diff
@@ -361,51 +252,69 @@ bool ParametersProcessor::processVariables(filamat::MaterialBuilder& builder,
 }
 
-bool ParametersProcessor::processRequires(filamat::MaterialBuilder& builder,
-        const JsonishValue& value) {
+static bool processRequires(MaterialBuilder& builder, const JsonishValue& value) {
+    static const std::unordered_map<std::string, filament::VertexAttribute> strToEnum {
+        { "color", filament::VertexAttribute::COLOR },
+        { "position", filament::VertexAttribute::POSITION },
+        { "tangents", filament::VertexAttribute::TANGENTS },
+        { "uv0", filament::VertexAttribute::UV0 },
+        { "uv1", filament::VertexAttribute::UV1 },
+    };
     for (auto v : value.toJsonArray()->getElements()) {
         if (v->getType() != JsonishValue::Type::STRING) {
-            std::cerr << PARAM_KEY_REQUIRES << ": entries must be STRINGs." << std::endl;
+            std::cerr << "requires: entries must be STRINGs." << std::endl;
             return false;
         }
 
         auto jsonString = v->toJsonString();
-        if (!isStringValidEnum(mStringToAttributeIndex, jsonString->getString())) {
-            return logEnumIssue(PARAM_KEY_REQUIRES, *jsonString, mStringToAttributeIndex);
+        if (!isStringValidEnum(strToEnum, jsonString->getString())) {
+            return logEnumIssue("requires", *jsonString, strToEnum);
         }
 
-        builder.require(stringToEnum(mStringToAttributeIndex, jsonString->getString()));
+        builder.require(stringToEnum(strToEnum, jsonString->getString()));
     }
 
     return true;
 }
 
-bool ParametersProcessor::processBlending(filamat::MaterialBuilder& builder,
-        const JsonishValue& value) {
+static bool processBlending(MaterialBuilder& builder, const JsonishValue& value) {
+    static const std::unordered_map<std::string, MaterialBuilder::BlendingMode> strToEnum {
+        { "add", MaterialBuilder::BlendingMode::ADD },
+        { "masked", MaterialBuilder::BlendingMode::MASKED },
+        { "opaque", MaterialBuilder::BlendingMode::OPAQUE },
+        { "transparent", MaterialBuilder::BlendingMode::TRANSPARENT },
+        { "fade", MaterialBuilder::BlendingMode::FADE },
+    };
     auto jsonString = value.toJsonString();
-    if (!isStringValidEnum(mStringToBlendingMode, jsonString->getString())) {
-        std::cerr << PARAM_KEY_BLENDING << ": value is not a valid BlendMode." << std::endl;
+    if (!isStringValidEnum(strToEnum, jsonString->getString())) {
+        std::cerr << "blending: value is not a valid BlendMode." << std::endl;
         return false;
     }
 
-    builder.blending(stringToEnum(mStringToBlendingMode, jsonString->getString()));
+    builder.blending(stringToEnum(strToEnum, jsonString->getString()));
     return true;
 }
 
-bool ParametersProcessor::processPostLightingBlending(filamat::MaterialBuilder& builder,
-        const JsonishValue& value) {
+static bool processPostLightingBlending(MaterialBuilder& builder, const JsonishValue& value) {
+    static const std::unordered_map<std::string, MaterialBuilder::BlendingMode> strToEnum {
```

Comment:
Could you remove "fade" and "masked" from this map so we don't need the switch statement a few lines below?

Reply:
Doh good idea!  Done.